### ISSUE TITLE
Question about Deblur-NeRF benchmark scenes and dataset training instructions

### ISSUE THREAD

**Issue by songwenhao123** [PotatoBigRoom/CoherentGS#2]
Thank you for sharing your excellent work and for releasing the code.

While reading the paper, I noticed the following statement:

“We evaluate our method on the standard Deblur-NeRF benchmark [2], which comprises 5 synthetic and 5 real-world scenes characterized by severe motion blur.”

However, I could not find which specific 5 synthetic scenes and which 5 real-world scenes were used. I also could not locate a script or step-by-step instructions in the repository for training on the Deblur-NeRF dataset (e.g., downloading/preprocessing the dataset, expected directory structure, and the exact training commands).

Could you please clarify:

What are the exact scene names/identifiers for the 5 synthetic and 5 real-world scenes used in your experiments?

Is there a recommended dataset preparation pipeline (download link, preprocessing steps, folder structure)?

Do you have an example training script/command (or config file) that reproduces the Deblur-NeRF benchmark results?

Any guidance (even a brief pointer to the relevant commit/config) would be greatly appreciated.

**Comment by PotatoBigRoom** [comment]
Hello, thank you for your interest in our project. You can find the download instructions for the Deblur-NeRF dataset at https://github.com/WU-CVGL/BAD-Gaussians. For the specific names of the 5 synthetic scenes and 5 real-world scenes used, please refer to the supplementary materials section of our paper.

**Comment by WenhaoSong-1512** [comment]
Hello, thank you for the impressive work on CoherentGS! While studying the implementation of the virtual view generation in `process_virtual_views_batch`, I noticed a potential discrepancy between the sampling strategy described in the paper (SACN) and the actual code execution logic. In the default configuration, `virtual_view_poses_per_step` is set to 2. However, the loop for generating interpolated frames relies on fixed indices (`i==0` and `i==1`), which makes the random exploration block unreachable.
```python
for i in range(cfg.virtual_view_poses_per_step):
                # 交替生成前向和后向插值
                if i == 0:
                    # Forward interpolation: between training views
                    interpolated_pose = interpolated_pose_forward
                    direction = "forward"
                    alpha = forward_alpha
                elif i == 1 and use_backward_interpolation:
                    # Backward interpolation: explore outside virtual views
                    interpolated_pose = interpolated_pose_backward
                    direction = "backward"
                    alpha = backward_alpha
                elif i == 1 and not use_backward_interpolation:
                    # 如果后向插值不可用，跳过
                    continue
                else:
                    # For more than 2, randomly choose interpolation strategy
                    if torch.rand(1).item() < 0.5:
                        # Between training views
                        alpha = torch.rand(1).item() * 0.8 + 0.1
                        interpolated_pose, _ = se3_interpolate_to_target(
                            train_pose1, train_K1, 
                            train_pose2, train_K2, 
                            t=alpha
                        )
                        direction = "forward-random"
                    elif use_backward_interpolation:
                        # Explore outside virtual views (random direction)
                        if torch.rand(1).item() < 0.5:
                            # Forward exploration: t > 1.0
                            alpha = torch.rand(1).item() * 0.5 + 1.0  # [1.0, 1.5]
                            interpolated_pose, _ = se3_interpolate_to_target(
                                virtual_pose1, virtual_K1, 
                                virtual_pose2, virtual_K2, 
                                t=alpha
                            )
                            direction = "backward-random-forward"
                        else:
                            # Backward exploration: t < 0.0
                            alpha = torch.rand(1).item() * 0.5 - 0.5  # [-0.5, 0.0]
                            interpolated_pose, _ = se3_interpolate_to_target(
                                virtual_pose2, virtual_K2, 
                                virtual_pose1, virtual_K1, 
                                t=alpha
                            )
                            direction = "backward-random-backward"
                    else:
                        # If backward interpolation unavailable, use forward interpolation
                        alpha = torch.rand(1).item() * 0.8 + 0.1
                        interpolated_pose, _ = se3_interpolate_to_target(
                            train_pose1, train_K1, 
                            train_pose2, train_K2, 
                            t=alpha
                        )
                        direction = "forward-random"
                
                print(f"    Generate {direction} interpolation frame (α={alpha:.3f})")
```
cfg.virtual_view_poses_per_step = 2, Since i never exceeds 1, the else branch containing the randomized alpha sampling is never executed. Looking forward to your insights. Thanks!

**Comment by PotatoBigRoom** [comment]
Thank you for your interest in our project. I sincerely apologize for any confusion you may have experienced during your learning. Firstly, random exploration is reflected in `virtual_indices = torch.randperm(len(self.available_interpolation_views))[:2]`, which selects back-interpolated frames, while for forward-interpolated frames, `alpha = torch.rand(1).item() * 0.8 + 0.1` Thank you for your question; I have fixed the code.

**Comment by WenhaoSong-1512** [comment]
Thank you again for your previous assistance regarding the random interpolation fix. After applying the fix for the forward interpolation logic, I proceeded to run the evaluation on 3-view sparse setting. However, I noticed that the reproduced results are still slightly lower than the metrics reported in the paper/supplementary materials.
**Results Comparison (Scene 001, 3-views)**

Here is the comparison between my reproduction (at step 6999) and the reported values:

| Metric | My Reproduction | Paper / Reported | Gap |
| :--- | :--- | :--- | :--- |
| **PSNR** | 17.95 | 18.26 | -0.31 |
| **SSIM** | **0.599** | **0.657** | **-0.058** |
| **LPIPS**| 0.348 | 0.322 | +0.026 |

**Detailed Log**
Here is the content of my `nvs_step6999.json`:
```json
{
  "psnr": 17.95183753967285,
  "ssim": 0.5988978743553162,
  "lpips": 0.3475738763809204,
  "best_psnr": 20.6086483001709,
  "best_ssim": 0.686124324798584,
  "best_lpips": 0.23246347904205322,
  "psnr_per_sample": [
    14.740002632141113,
    20.6086483001709,
    19.888824462890625,
    18.358076095581055,
    16.163633346557617
  ],
  "ssim_per_sample": [
    0.5699737668037415,
    0.686124324798584,
    0.6520184874534607,
    0.5837574005126953,
    0.5026154518127441
  ],
  "lpips_per_sample": [
    0.46759551763534546,
    0.23246347904205322,
    0.24540628492832184,
    0.37008798122406006,
    0.42231616377830505
  ]
}
I have observed similar performance gaps across other scenes as well. Could you please advise if there are specific hyperparameters that need to be tuned? Any insights would be greatly appreciated. Thank you!
